ThreadOfNineLives.IntegrationTests/MiddlewareIntegrationTests.cs

```diff
@@ -0,0 +1,227 @@
+﻿using Moq;
+using Backend.Repositories;
+using Backend.Services;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace ThreadOfNineLives.IntegrationTests
+{
+    public class MiddlewareIntegrationTests : IClassFixture<WebApplicationFactory<Backend.Program>>
+    {
+        private readonly WebApplicationFactory<Backend.Program> _factory;
+
+        public MiddlewareIntegrationTests(WebApplicationFactory<Backend.Program> factory)
+        {
+            // Setup the factory with mocked services
+            _factory = factory.WithWebHostBuilder(builder =>
+            {
+                builder.ConfigureServices(services =>
+                {
+                    // Remove the existing IUserRepository registration if any
+                    var userDescriptor = services.SingleOrDefault(d => d.ServiceType == typeof(IUserRepository));
+                    if (userDescriptor != null)
+                    {
+                        services.Remove(userDescriptor);
+                    }
```

Comment:
Logikken her med at fjerne repoet er rigtigt fin, men logisk set kan der ske en fejl her. Fordi der bliver brugt singleOrDefault() så fjerner den kun en hvis der er præcis 1. Hvis der er kommet til at blive registreret flere (kommer nok ikke til at ske, men for at gå efter den logik der er blevet beskrevet) så tror jeg vi burde se om vi kan fjerne alle implementationer af IUserRepository.


Forestiller mig at vi skal lave noget ala:

 var userDescriptors = services.Where(d => d.ServiceType == typeof(IUserRepository));
                    if (userDescriptor.count() != 0)
                    {
                        services.RemoveRange(userDescriptors);
                    }

Reply:
Synes også vi skal prøve at beholde de her tests, men se om vi kan omstrukturere så de er i de respektive apitest og unit tests fremover.